torch/csrc/jit/codegen/cuda/interface.cpp

```diff
@@ -26,6 +26,32 @@ namespace jit {
 namespace fuser {
 namespace cuda {
 
+#ifdef USE_CUDA
+class LoadingNvfuserLibrary {
```

Comment:
I think this doesn't work internally:
- I don't think USE_CUDA is defined in the cuda builds? (cc @malfet, is this true? I remember running into this in the past with cudnn jit issues, I don't remember the resolution)
- We probably would need to add new buck targets for libnvfuser_codegen.so, and idk if "looking for a .so file at runtime" will work with buck?

fwiw, it appears that currently this fails silently internally (`setEnabled(true)` doesn't error out, but nvfuser doesn't run)

Reply:
Silent error is bad. I'll add an API to query whether nvfuser is loaded and inside setEnabled we'll check for it so at least it's not a silent error.

>    I don't think USE_CUDA is defined in the cuda builds? (cc @malfet, is this true? I remember running into this in the past with cudnn jit issues, I don't remember the resolution)

this is a surprise. Should I be looking for a different flag? We have USE_CUDA everywhere in the code tho, even under torch/... Wondering if there's a preprocessing or something that translates it shrug

>    We probably would need to add new buck targets for libnvfuser_codegen.so, and idk if "looking for a .so file at runtime" will work with buck?

So here's really two problems here:

1. buck targets for libnvfuser_codegen.so
    I understand it as buck relies on bazel file to build? I don't know if there's any extra hook to put it together tho.
    I do have a TODO to enable bazel in a following PR. We can drag that into this PR if we absolutely need bazel to work as well.

2. if "looking for a .so file at runtime" will work with buck?
    You guys are definitely the expert with your internal workflow. I'll follow your advise on how to proceed here.

    I was hoping it would work since we used it for nvrtc as well: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cuda/detail/CUDAHooks.cpp#L174-L187

    But there is the USE_DIRECT_NVRTC macro that you guys might be using for internal workload.

    If dlopen is out of the window, is there an alternative for dynamically loading libraries for internal workflow?